File: src/Web/Holefeeder.Web/ClientApp/src/app/shared/enums/account-type.enum.ts
export enum AccountType {
  checking = 'Checking',
  creditCard = 'CreditCard',
  creditLine = 'CreditLine',
  investment = 'Investment',
  loan = 'Loan',
  mortgage = 'Mortgage',
  savings = 'Savings'
}

export const AccountTypeNames = new Map<string, string>([
  [AccountType.checking, 'Checking'],
  [AccountType.creditCard, 'Credit card'],
  [AccountType.creditLine, 'Credit line'],
  [AccountType.investment, 'Investment'],
  [AccountType.loan, 'Loan'],
  [AccountType.mortgage, 'Mortgage'],
  [AccountType.savings, 'Savings']
]);

export function accountTypeMultiplier(type: AccountType): number {
  switch (type) {
    case AccountType.checking:
    case AccountType.investment:
    case AccountType.savings:
      return 1;
    default:
      return -1;
  }
}

File: src/Holefeeder.Ui/lib/core/enums/account_type_enum.dart
enum AccountType {
  checking,
  creditCard,
  creditLine,
  investment,
  loan,
  mortgage,
  savings,
}

extension AccountTypeExtension on AccountType {
  static AccountType fromString(String type) {
    return AccountType.values.firstWhere(
      (e) => e.toString().split('.').last.toLowerCase() == type.toLowerCase(),
      orElse: () => throw ArgumentError('Invalid AccountType: $type'),
    );
  }

  String toStringValue() {
    return toString().split('.').last;
  }
}
